### ISSUE TITLE
error TS2515: Non-abstract class 'PuppetXp' does not implement inherited abstract member 'messageLocation' 

### ISSUE THREAD

**Issue by printf172** [wechaty/wechaty-puppet-xp#30]
Xx:
pcWx : 3.3.0.115 
node : v16.10.0
npm : 7.24.0
mac windows10虚拟机

PS C:\work\wechaty-puppet-xp> npm start

> wechaty-puppet-xp@0.7.12 start
> cross-env NODE_OPTIONS="--no-warnings --loader=ts-node/esm" node examples/ding-dong-bot.ts

C:\work\wechaty-puppet-xp\node_modules\ts-node\src\index.ts:692
    return new TSError(diagnosticText, diagnosticCodes);
           ^
TSError: ⨯ Unable to compile TypeScript:
src/puppet-xp.ts:67:7 - error TS2515: Non-abstract class 'PuppetXp' does not implement inherited abstract member 'messageLocation' from class 'Puppet'.

67 class PuppetXp extends Puppet {
         ~~~~~~~~
src/puppet-xp.ts:67:7 - error TS2515: Non-abstract class 'PuppetXp' does not implement inherited abstract member 'messageSendLocation' from class 'Puppet'.

67 class PuppetXp extends Puppet {
         ~~~~~~~~

    at createTSError (C:\work\wechaty-puppet-xp\node_modules\ts-node\src\index.ts:692:12)
    at reportTSError (C:\work\wechaty-puppet-xp\node_modules\ts-node\src\index.ts:696:19)
    at getOutput (C:\work\wechaty-puppet-xp\node_modules\ts-node\src\index.ts:883:36)
    at Object.compile (C:\work\wechaty-puppet-xp\node_modules\ts-node\src\index.ts:1185:30)
    at C:\work\wechaty-puppet-xp\node_modules\ts-node\src\esm.ts:146:38
    at Generator.next (<anonymous>)
    at C:\work\wechaty-puppet-xp\node_modules\ts-node\dist\esm.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (C:\work\wechaty-puppet-xp\node_modules\ts-node\dist\esm.js:4:12)
    at transformSource (C:\work\wechaty-puppet-xp\node_modules\ts-node\dist\esm.js:88:16) {
  diagnosticText: "\x1B[96msrc/puppet-xp.ts\x1B[0m:\x1B[93m67\x1B[0m:\x1B[93m7\x1B[0m - \x1B[91merror\x1B[0m\x1B[90m TS2515: \x1B[0mNon-abstract class 'PuppetXp' does not implement inherited abstract member 'messageLocation' from class 'Puppet'.\r\n" +
    '\r\n' +
    '\x1B[7m67\x1B[0m class PuppetXp extends Puppet {\r\n' +
    '\x1B[7m  \x1B[0m \x1B[91m      ~~~~~~~~\x1B[0m\r\n' +
    "\x1B[96msrc/puppet-xp.ts\x1B[0m:\x1B[93m67\x1B[0m:\x1B[93m7\x1B[0m - \x1B[91merror\x1B[0m\x1B[90m TS2515: \x1B[0mNon-abstract class 'PuppetXp' does not implement inherited abstract member 'messageSendLocation' from class 'Puppet'.\r\n" +
    '\r\n' +
    '\x1B[7m67\x1B[0m class PuppetXp extends Puppet {\r\n' +
    '\x1B[7m  \x1B[0m \x1B[91m      ~~~~~~~~\x1B[0m\r\n',
  diagnosticCodes: [ 2515, 2515 ]
}

**Comment by huan** [comment]
Sorry for this inconvenience, this breaking changes introduced by the wechaty puppet update yesterday.

My suggestion would be use the https://github.com/wechaty/wechaty-getting-started to load XP puppet and it should be OK.

Otherwise if you'd like to use this repo with TypeScript, Pull Request would be welcome, or you can wait us to fix it later.

**Comment by huan** [comment]
Should be fixed by `wechaty-puppet-xp@0.7.13`

**Comment by atorber** [comment]
最新代码提示Type 'PuppetXp' is not assignable to type

```
examples/ripe-wechaty.ts:50:3 - error TS2322: Type 'PuppetXp' is not assignable to type '"wechaty-puppet-hostie" | "wechaty-puppet-puppeteer" | "wechaty-puppet-service" | "wechaty-puppet-mock" | "wechaty-puppet-wechat" | "wechaty-puppet-wechat4u" | "wechaty-puppet-padlocal" | ... 9 more ... | undefined'.
  Type 'PuppetXp' is not assignable to type 'Puppet'.
    Property 'options' is protected but type 'PuppetXp' is not a class derived from 'Puppet'.

50   puppet,
     ~~~~~~

tests/integration.spec.ts:13:33 - error TS2322: Type 'PuppetXp' is not assignable to type '"wechaty-puppet-hostie" | "wechaty-puppet-puppeteer" | "wechaty-puppet-service" | "wechaty-puppet-mock" | "wechaty-puppet-wechat" | "wechaty-puppet-wechat4u" | "wechaty-puppet-padlocal" | ... 9 more ... | undefined'.

13   const wechaty = new Wechaty({ puppet })
```
